src/lightning/fabric/wrappers.py

```diff
@@ -322,5 +349,28 @@ def is_wrapped(obj: object) -> bool:
         obj: The object to test.
 
     """
-    obj = _unwrap_compiled(obj)
+    obj, _ = _unwrap_compiled(obj)
     return isinstance(obj, (_FabricModule, _FabricOptimizer, _FabricDataLoader))
+
+
+def _capture_compile_kwargs(compile_fn: Callable) -> Callable:
```

Comment:
To be exhaustive, you would also need to capture the global dynamo/inductor state at this point in time and restore it when compile is actually applied:

https://github.com/pytorch/pytorch/blob/main/torch/_dynamo/config.py
https://github.com/pytorch/pytorch/blob/main/torch/_inductor/config.py

Reply:
I would give a) a week and if nothing happens, go with c) and add comments about this limitation/unknown

Reply:
I added comments about the limitation in the function that extracts the arguments.